napari/_qt/_tests/test_qt_notifications.py

```diff
@@ -192,8 +192,14 @@ def test_notification_error(mock_show, monkeypatch, qtbot):
     except ValueError as e:
         notif = ErrorNotification(e)
 
-    dialog = NapariQtNotification.from_notification(notif)
-    qtbot.add_widget(dialog)
+    # This test creates TracebackDialog which is not added to qtbot
+    # but its parent is same as parent of NapariQtNotification.
+    # so create dummy parent allow to not leak widget.
+    widget = QWidget()
+    qtbot.add_widget(widget)
+
+    dialog = NapariQtNotification.from_notification(notif, parent=widget)
```

Comment:
Why does the dialog need a parent? From the [`QDialog`](https://doc.qt.io/qt-6/qdialog.html#QDialog) docs, it seems fine to have a dialog as a top level widget without a parent. And then [`qtbot`](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.addWidget) should handle closing it, which should also eventually free up memory.

Not saying you're wrong to do this, but I don't understand this change yet.

Reply:
The `NapariQtNotification` dialog doe not need to have a set parent, as it could be handled by `qtbot`, but without this, I do not know how to handle `TracebackDialog` to be guarded by `qtbot`

Reply:
Ah, I get it, the Qt parent hierarchy is:

```
QWidget
    NapariQtNotification
    TracebackDialog
```

Previously, we just tracked the instance of `NapariQtNotification`, which didn't have a parent, but that means `TracebackDialog` wasn't tracked by qtbot. So `parent` here is representing some main widget or window in napari.

Reply:
Yes